pkg/converter/source/virtualmachine_converter.go

```diff
@@ -155,12 +155,17 @@ func (v VMConverter) processEvent(vm *VirtualMachineSource, failedUpdates map[st
 	if !isNotFound {
 		if isAgent {
 			base := client.MergeFrom(externNode.DeepCopy())
-			patch := target.PatchExternalNodeFrom(vm, externNode, v.Client)
-			err = v.Client.Patch(ctx, patch, base)
+			patch, changed := target.PatchExternalNodeFrom(vm, externNode, v.Client)
+			if changed {
+				err = v.Client.Patch(ctx, patch, base)
+			}
```

Comment:
return early if not changed, so that no logs are written. Same for below L166

Reply:
both changed